README.md

```diff
@@ -194,483 +211,135 @@ CommandsBefore:
-the `Type:` in the `Demo` resource.
-
-```YAML
-# demo-custom-template.yaml
-$params:
-  - Name: Foo
-    Type: string
-  - Name: Bar
-    Type: string
-Resources:
-  Topic1:
-    Type: AWS::SNS::Topic
-    Properties:
-      DisplayName: "Hello {{Foo}}"
-  Topic2:
-    Type: AWS::SNS::Topic
-    Properties:
-      DisplayName: "Hello {{Bar}}"
+- CLI options (see above)
+- `AWS_REGION` and `AWS_PROFILE` environment variables
+- standard [AWS environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html)
+- `Region` and `Profile` or `AssumeRoleArn` in args file
 
-```
+These credentials are used to interact with the CloudFormation API and to
+perform and AWS-related imports.
 
-```YAML
-# demo-cfn-template.yaml
-$imports:
-  DemoTmpl: demo-custom-template.yaml
-  # ^ the name is arbitrary
-Resources:
-  Demo:
-    Type: DemoTmpl # use it here
-    Properties:
-      # these are threaded in as the $params defined in demo-custom-template.yaml
-      Foo: 123
-      Bar: 456
-  SomeOtherResource:
-    Type: AWS::SNS::Topic
-    Properties:
-      DisplayName: "non-custom"
-```
+If your profile requires an MFA token, iidy will prompt for it.
 
-We can use `iidy render demo-cfn-template.yaml` to see the final
-template:
-
-```YAML
-AWSTemplateFormatVersion: '2010-09-09'
-Resources:
-  DemoTopic1:
-    Type: 'AWS::SNS::Topic'
-    Properties:
-      DisplayName: Hello 123
-  DemoTopic2:
-    Type: 'AWS::SNS::Topic'
-    Properties:
-      DisplayName: Hello 456
-  SomeOtherResource:
-    Type: 'AWS::SNS::Topic'
-    Properties:
-      DisplayName: non-custom
-Metadata: # ... elided ...
 ```
-
-TODO: more examples and cover the details
-TODO: This section is best explained with an worked example / executable demonstration
-
-### Working with *Non*-CloudFormation YAML
-
-`iidy render` can be used to invoke the pre-processor on any YAML
-document. All features are available other than [`custom resource
-templates`](#custom-resource-templates). See `iidy render --help` for
-some more information.
-
-### Converting Deployed Stacks to `iidy`
-Converting current stacks to `iidy` can be done through the `iidy convert-stack-to-iidy` command,
-which accepts 2 parameters: `stackname` (the stack name of the project that you'd want to convert) and
-an `outputDir` (path where the output will be stored). For example:
-
-```Shell
-iidy convert-stack-to-iidy my-cloudformation-stack-1 ./infrastructure
-
-# this should generate 3 files:
-# * _original-template.json
-# * stack-args.yaml
-# * cfn-template.yaml
+? MFA token for arn:aws:iam::002682819933:mfa/example.user: ____
 ```
 
-### `iidy` Environment Variables
+If you've assumed a profile prior to running iidy and want to it to ignore
+what's specified as `Profile` in `stack-args.yaml` and instead use `AWS_*`
+environment variables, set the CLI option `--profile no-profile`.
 
-Any parameter used by iidy can be set using IIDY_{{argname}}
+## Additional Documentation
```

Comment:
I think we should lead with the readme with TOC of all documentation for easy reference.